### PR TITLE
Add DefaultSequence to the model

### LINKED ISSUE #1568: Add default sequence to the model when used
When I configure a model to use the default sequence, the default sequence should be added to the model. Here is an example of using the default sequence.

```C#
modelBuilder.Entity<Person>().Property(p => p.Id).ForSqlServer().UseSequence();
```

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #1568 
Also changed few missed nameof

### CODE REVIEW

File: src/EntityFramework.SqlServer/Metadata/ModelConventions/SqlServerValueGenerationStrategyConvention.cs [hunk 1]
```diff
@@ -70,10 +70,14 @@ public virtual InternalRelationshipBuilder Apply(InternalRelationshipBuilder rel
...
-                SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.ValueGeneration,
-                SqlServerValueGenerationStrategy.Sequence.ToString(),
-                ConfigurationSource.Convention);
+            var extensions = modelBuilder.Metadata.SqlServer();
+
+            var sequence = extensions.GetOrAddSequence();
+
+            extensions.ValueGenerationStrategy = SqlServerValueGenerationStrategy.Sequence;
+            extensions.DefaultSequenceName = sequence.Name;
+            extensions.DefaultSequenceSchema = sequence.Schema;
```
Comment (ajcvickers) [inline src/EntityFramework.SqlServer/Metadata/ModelConventions/SqlServerValueGenerationStrategyConvention.cs]: Doesn't this mean that these are no longer marked as being set by convention?

Comment (ajcvickers) [inline src/EntityFramework.SqlServer/Metadata/ModelConventions/SqlServerValueGenerationStrategyConvention.cs]: Also, the sequence should not be added to the model if it is not needed by the model. This is important for supporting SQL Server 2008 where creating the sequence will fail and so applications should be able to switch 100% to Identity and have no remnants of sequence functionality remaining.

File: src/EntityFramework.SqlServer/Metadata/ModelConventions/SqlServerValueGenerationStrategyConvention.cs [hunk 2]
```diff
@@ -74,6 +74,12 @@ public virtual InternalModelBuilder Apply(InternalModelBuilder modelBuilder)
                 SqlServerAnnotationNames.Prefix + SqlServerAnnotationNames.ValueGeneration,
                 SqlServerValueGenerationStrategy.Sequence.ToString(),
                 ConfigurationSource.Convention);
+
+            var extensions = modelBuilder.Metadata.SqlServer();
+            var sequence = extensions.GetOrAddSequence();
```
Comment (bricelam) [inline src/EntityFramework.SqlServer/Metadata/ModelConventions/SqlServerValueGenerationStrategyConvention.cs]: The sequence annotation should be marked as "by convention" and removed if the user calls `.UseIdentity()`. We should also file a bug to remove it if it never ends up being used.

**Comment by smitpatel** [comment]
filed #1994

File: src/EntityFramework.SqlServer/Metadata/ModelConventions/SqlServerValueGenerationStrategyConvention.cs [hunk 3]
```diff
@@ -15,6 +15,11 @@ public virtual InternalModelBuilder Apply(InternalModelBuilder modelBuilder)
                 SqlServerValueGenerationStrategy.Sequence.ToString(),
                 ConfigurationSource.Convention);
 
+            var extensions = modelBuilder.Metadata.SqlServer();
+            var sequence = extensions.GetOrAddSequence();
+            extensions.DefaultSequenceName = sequence.Name;
+            extensions.DefaultSequenceSchema = sequence.Schema;
```
Comment (bricelam) [inline src/EntityFramework.SqlServer/Metadata/ModelConventions/SqlServerValueGenerationStrategyConvention.cs]: Should these somehow be passed `ConfigurationSource.Convention`?

File: src/EntityFramework.SqlServer/Metadata/SqlServerModelBuilder.cs [hunk 4]
```diff
@@ -20,7 +20,13 @@ public SqlServerModelBuilder([NotNull] Model model)
 
         public virtual SqlServerModelBuilder UseIdentity()
         {
-            _model.SqlServer().ValueGenerationStrategy = SqlServerValueGenerationStrategy.Identity;
+            var extensions = _model.SqlServer();
+
+            extensions.RemoveSequence();
```
Comment (bricelam) [inline src/EntityFramework.SqlServer/Metadata/SqlServerModelBuilder.cs]: We should only do this if it was added by convention.

Comment (smitpatel) [inline src/EntityFramework.SqlServer/Metadata/SqlServerModelBuilder.cs]: at Model level it is not possible to know configuration source of the annotation. So we cannot verify if this was added by convention or not.

**Comment by bricelam** [comment]
:shipit:

**Pull request merged**